integration-tests/tests/utils/config.py

```diff
@@ -0,0 +1,50 @@
+import shutil
+from dataclasses import dataclass
+from pathlib import Path
+
+
+@dataclass(frozen=True)
+class PackageConfig:
+    clp_bin_dir: Path
+    clp_package_dir: Path
+    clp_sbin_dir: Path
+    test_output_dir: Path
+    uncompressed_logs_dir: Path
+
+
+@dataclass(frozen=True)
+class DatasetLogs:
+    dataset_name: str
+    tarball_url: str
+    download_dir: Path
+    extraction_dir: Path
+    compression_dir: Path
+    decompression_dir: Path
+
+    @classmethod
+    def create(cls, package_config, dataset_name: str, **kwargs):
+        dataset_name = dataset_name.strip()
+        if 0 == len(dataset_name):
+            raise ValueError("`dataset_name` cannot be empty.")
+        return cls(
+            dataset_name=dataset_name,
+            tarball_url=kwargs.get("tarball_url", ""),
+            download_dir=kwargs.get(
+                "download_dir", package_config.uncompressed_logs_dir / f"{dataset_name}.tar.gz"
+            ),
+            extraction_dir=kwargs.get(
+                "extraction_dir", package_config.uncompressed_logs_dir / dataset_name
+            ),
+            compression_dir=kwargs.get(
+                "compression_dir", package_config.test_output_dir / f"{dataset_name}-archives"
+            ),
+            decompression_dir=kwargs.get(
+                "decompression_dir",
+                package_config.test_output_dir / f"{dataset_name}-decompressed-logs",
+            ),
```

Comment:
Just thinking, would it make sense to have separate folder for each dataset? so the structure will look like

test_root_dir
- dataset_dir
- - downloaded_logs
- - archives
- - decompressed_logs
....

Reply:
I think we can leave dataset support to future PRs. Also, I don't like the name conflicts between `dataset_logs` and the dataset feature we just recently added.